### PR TITLE
(Github Action) Upload library to S3

### PR DESCRIPTION
**Pull request by PabloReszczynski**
This configuration should, on a tag push:

- Run tests
- Build the library
- Upload it to S3
- Revalidate the cloudfront cache

Blocked by: https://github.com/Topsort/scripts/pull/363

### CODE REVIEW

File: .github/workflows/npm-publish.yml [hunk 1]
```diff
@@ -41,3 +41,38 @@ jobs:
...
+          restore-keys: |
+            ${{ runner.os }}-node-
+
+      - name: Build
+        run: npm run build
+
+      - name: Upload to S3
+        uses: lbertenasco/s3-deploy@v1
+        with:
+            folder: dist
```
[the remark this answers is not in the document]
Comment (PabloReszczynski) [reply .github/workflows/npm-publish.yml]: done

File: .github/workflows/npm-publish.yml [hunk 2]
```diff
@@ -41,3 +41,38 @@ jobs:
         env: # More info about the environment variables in the README
           GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }} # Leave this as is, it's automatically generated
           NPM_AUTH_TOKEN: ${{ secrets.NPM_AUTH_TOKEN }} # You need to set this in your repo settings
+
+  upload-to-s3:
+    name: upload-to-s3
+    runs-on: ubuntu-latest
+    env:
+      AWS_ACCESS_KEY_ID: ${{ secrets.AWS_ACCESS_KEY_ID }}
+      AWS_SECRET_ACCESS_KEY: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
```
[the remark this answers is not in the document]
Comment (PabloReszczynski) [reply .github/workflows/npm-publish.yml:50]: I'm will ask the infra team for a aws access key with only access to that particular S3 bucket and cloudfront id

File: .github/workflows/npm-publish.yml [hunk 3]
```diff
@@ -41,3 +41,38 @@ jobs:
         env: # More info about the environment variables in the README
           GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }} # Leave this as is, it's automatically generated
           NPM_AUTH_TOKEN: ${{ secrets.NPM_AUTH_TOKEN }} # You need to set this in your repo settings
+
```
[the remark this answers is not in the document]
Comment (PabloReszczynski) [reply .github/workflows/npm-publish.yml:44]: I've changed the action so it only runs on a tag push, so we should do the version push manually before making a new release

**Comment by PabloReszczynski** [comment]
By suggestion of @bgreen-topsort, I am reimplemeting this with Codebuild instead

**Review by sk- — APPROVED** [review]

**Pull request merged**